athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBRecordHandler.java

```diff
@@ -138,66 +139,45 @@ protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recor
-                    // had not made any DDB calls yet and there may be zero items returned when it does
-                    return 0;
-                }
 
-                boolean matched = true;
-                numResultRows.getAndIncrement();
-                // TODO refactor to use GeneratedRowWriter to improve performance
-                for (Field nextField : recordsRequest.getSchema().getFields()) {
-                    Object value = ItemUtils.toSimpleValue(item.get(nextField.getName()));
-                    Types.MinorType fieldType = Types.getMinorTypeForArrowType(nextField.getType());
-
-                    value = DDBTypeUtils.coerceValueToExpectedType(value, nextField, fieldType, recordMetadata);
-
-                    try {
-                        switch (fieldType) {
-                            case LIST:
-                                // DDB may return Set so coerce to List. Also coerce each List item to the correct type.
-                                List valueAsList = value != null
-                                        ? DDBTypeUtils.coerceListToExpectedType(value, nextField, recordMetadata) : null;
-                                matched &= block.offerComplexValue(nextField.getName(),
-                                        rowNum,
-                                        resolver,
-                                        valueAsList);
-                                break;
-                            case STRUCT:
-                                matched &= block.offerComplexValue(nextField.getName(),
-                                        rowNum,
-                                        resolver,
-                                        value);
-                                break;
-                            default:
-                                matched &= block.offerValue(nextField.getName(), rowNum, value);
-                                break;
-                        }
-
-                        if (!matched) {
-                            return 0;
-                        }
-                    }
-                    catch (Exception ex) {
-                        throw new RuntimeException("Error while processing field " + nextField.getName(), ex);
-                    }
-                }
-                return 1;
-            });
+            Map<String, AttributeValue> item = itemIterator.next();
+            if (item == null) {
+                // this can happen regardless of the hasNext() check above for the very first iteration since itemIterator
+                // had not made any DDB calls yet and there may be zero items returned when it does
+                continue;
+            }
+            numRows++;
+            try {
+                spiller.writeRows((Block block, int rowNum) -> rowWriter.writeRow(block, rowNum, item) ? 1 : 0);
+            }
+            catch (Exception ex) {
+                logger.warn("Exception while writing rows :{}", ex.getMessage(), ex);
```

Comment:
Is it safe to continue to write the other rows when this exception happens?

Also what about `numRows`, it was incremented before this exception happened, should `numRows` be incremented inside the `try{}` after `writeRows` succeeds instead?